dss/operations/elasticsearch.py

```diff
@@ -26,18 +27,40 @@
 elasticsearch = dispatch.target("elasticsearch", help=__doc__)
 
 
-@elasticsearch.action("index",
+@elasticsearch.action("index-keys",
                       arguments={"--replica": dict(choices=[r.name for r in Replica], required=True),
                                  "--keys": dict(default=None, nargs="*", help="keys to index.")})
-def index(argv: typing.List[str], args: argparse.Namespace):
+def index_keys(argv: typing.List[str], args: argparse.Namespace):
     """
-    Initiate the indexer lambda for `keys`
+    Queue an SQS message to the indexer lambda for each key in `keys`.
     """
     index_queue_url = get_queue_url("dss-index-operation-" + os.environ['DSS_DEPLOYMENT_STAGE'])
     with SQSMessenger(index_queue_url) as sqsm:
         for key in args.keys:
             sqsm.send(json.dumps(dict(replica=args.replica, key=key)))
 
+@elasticsearch.action("index",
+                      arguments={"--replica": dict(choices=[r.name for r in Replica], required=True),
+                                 "--prefix": dict(default="", help="UUID prefix to index.")})
+def index(argv: typing.List[str], args: argparse.Namespace):
+    """
+    Queue an SQS message to the indexer lambda for each key in object storage beginning with `bundles/{prefix}`.
+    If `prefix` is omitted, send a message for each key in object storage beginning with `bundles/`
+    """
+    replica = Replica[args.replica]
+    handle = Config.get_blobstore_handle(replica)
+    index_queue_url = get_queue_url("dss-index-operation-" + os.environ['DSS_DEPLOYMENT_STAGE'])
+
+    def _forward_keys(pfx):
+        with SQSMessenger(index_queue_url) as sqsm:
+            for key in handle.list(replica.bucket, pfx):
+                sqsm.send(json.dumps(dict(replica=args.replica, key=key)))
+
+    with ThreadPoolExecutor(max_workers=10) as e:
+        futures = [e.submit(_forward_keys, f"bundles/{args.prefix}{c}") for c in hexdigits.lower()]
```

Comment:
```
In [4]: hexdigits.lower()
Out[4]: '0123456789abcdefabcdef'
```
I dont think it would matter, but if the prefix is omitted, the queue would fill up 2x for the `a-e` chars. I'm not sure how it would be handled by the SQS Queue Consumer, but just wanted to note it.

Reply:
thanks for catching that! Will fix.